src/app/Core/adaptIsolatesFromAPI.service.ts

```diff
@@ -21,6 +22,7 @@ export function adaptIsolatesFromAPI(isolateProp: IsolateDTO): DbCollection {
             origin,
             category,
             productionType,
+            resistance: Object.keys(resistance) as ResistantValues[]

```

Comment:
Reminder: the type conversion here will not prevent the running code to add faulty values to resistance because the DTO data comes from an external source and cannot be checked during compile time. The code will fail elsewhere when you try to access the resistance object. So a more robust way would be:
```
possibleResistancesArray.foreach(res => { 
  if(dto.resistance.res !== undefined) {
    model.resistance.push(res)
  }
})
```

Reply:
Based on previous CR, there is already a tick for data validation. However, this does not have a high priority at the moment.